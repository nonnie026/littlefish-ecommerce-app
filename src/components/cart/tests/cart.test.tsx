import '@testing-library/jest-dom'
import { fireEvent, render, screen } from '@testing-library/react'
import { Cart } from '../index'

describe('Cart', () => {
  const mockHandleRemove = jest.fn()
  const cartItems = [
    {
      id: 1,
      title: 'Item 1',
      quantity: 2,
      price: 9.99,
      image: 'link-to-image',
      category: 'category',
      description: 'description',
      rating: {
        rate: 4.5,
        count: 10
      }
    }
  ]

  const mockAnchorEl = document.createElement('div')

  const view = () =>
    render(
      <Cart
        cartItems={cartItems}
        numberOfItems={3}
        total={29.97}
        id="cart-popover"
        anchorEl={mockAnchorEl}
        handleClose={() => {
          console.log('close')
        }}
        open={true}
        handleRemoveFromCart={mockHandleRemove}
      />
    )

  afterEach(() => {
    jest.clearAllMocks()
  })

  it('renders snapshot', () => {
    const { asFragment } = view()

    expect(asFragment()).toMatchSnapshot()
  })

  it('renders correctly with items', () => {
    view()
    cartItems.forEach((item) => {
      expect(screen.getByText(item.title)).toBeInTheDocument()
      expect(screen.getByText(`${item.quantity} x`)).toBeInTheDocument()
    })
  })

  it('renders call mockHandleRemove once', () => {
    view()
    fireEvent.click(screen.getByTestId('remove-item'))
    expect(mockHandleRemove).toHaveBeenCalledTimes(1)
  })
})
