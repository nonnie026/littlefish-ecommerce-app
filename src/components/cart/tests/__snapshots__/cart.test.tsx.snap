// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart renders correctly with no items 1`] = `<DocumentFragment />`;

exports[`Cart renders snapshot 1`] = `<DocumentFragment />`;
